integration/commands_diagnostic_test.go

```diff
@@ -271,38 +271,77 @@ func TestCommandsDiagnosticListCommands(t *testing.T) {
 
 func TestCommandsDiagnosticValidate(t *testing.T) {
 	t.Parallel()
+
 	ctx, collection := setup.Setup(t, shareddata.Doubles)
 
-	var doc bson.D
-	err := collection.Database().RunCommand(ctx, bson.D{{"validate", collection.Name()}}).Decode(&doc)
-	require.NoError(t, err)
+	t.Run("Basic", func(t *testing.T) {
+		t.Parallel()
+
+		var doc bson.D
+		command := bson.D{{"validate", collection.Name()}}
+		err := collection.Database().RunCommand(ctx, command).Decode(&doc)
+		require.NoError(t, err)
+
+		actual := ConvertDocument(t, doc)
+		expected := must.NotFail(types.NewDocument(
+			"ns", "TestCommandsDiagnosticValidate.TestCommandsDiagnosticValidate",
+			"nInvalidDocuments", int32(0),
+			"nNonCompliantDocuments", int32(0),
+			"nrecords", int32(25),
+			"nIndexes", int32(1),
+			"valid", true,
+			"repaired", false,
+			"warnings", types.MakeArray(0),
+			"errors", types.MakeArray(0),
+			"extraIndexEntries", types.MakeArray(0),
+			"missingIndexEntries", types.MakeArray(0),
+			"corruptRecords", types.MakeArray(0),
+			"ok", float64(1),
+		))
+
+		actual.Remove("keysPerIndex")
+		actual.Remove("indexDetails")
+		actual.Remove("$clusterTime")
+		actual.Remove("operationTime")
+
+		testutil.AssertEqual(t, expected, actual)
+	})
 
-	t.Log(doc.Map())
-
-	actual := ConvertDocument(t, doc)
-	expected := must.NotFail(types.NewDocument(
-		"ns", "TestCommandsDiagnosticValidate.TestCommandsDiagnosticValidate",
-		"nInvalidDocuments", int32(0),
-		"nNonCompliantDocuments", int32(0),
-		"nrecords", int32(0), // replaced below
-		"nIndexes", int32(1),
-		"valid", true,
-		"repaired", false,
-		"warnings", types.MakeArray(0),
-		"errors", types.MakeArray(0),
-		"extraIndexEntries", types.MakeArray(0),
-		"missingIndexEntries", types.MakeArray(0),
-		"corruptRecords", types.MakeArray(0),
-		"ok", float64(1),
-	))
-
-	actual.Remove("keysPerIndex")
-	actual.Remove("indexDetails")
-	actual.Remove("$clusterTime")
-	actual.Remove("operationTime")
-
-	testutil.CompareAndSetByPathNum(t, expected, actual, 39, types.NewStaticPath("nrecords"))
```

Comment:
I removed this it seem overly complex when we assume that `shareddata` is static.